### PR TITLE
Add loading variant for Snaps UI Button

### PR DESCRIPTION
**Pull request by david0xd**
This PR adds loading variant for Snap UI button. This button shows loading icon with an infinite animation.

Related ticket: https://github.com/MetaMask/snaps/issues/2694
Related extension PR: https://github.com/MetaMask/metamask-extension/pull/28997

Loading button in action:

https://github.com/user-attachments/assets/a2f15262-dc68-44cb-bbd2-22e288d9d09c

### CODE REVIEW

File: packages/snaps-sdk/src/jsx/validation.ts [hunk 1]
```diff
@@ -243,7 +243,9 @@ export const ButtonStruct: Describe<ButtonElement> = element('Button', {
   children: children([StringElementStruct, ImageStruct, IconStruct]),
   name: optional(string()),
   type: optional(nullUnion([literal('button'), literal('submit')])),
-  variant: optional(nullUnion([literal('primary'), literal('destructive')])),
+  variant: optional(
+    nullUnion([literal('primary'), literal('destructive'), literal('loading')]),
+  ),
   disabled: optional(boolean()),
```
Comment (FrederikBolding) [inline packages/snaps-sdk/src/jsx/validation.ts:249]: Hmm, should loading be a prop similar to `disabled` instead of a variant?

WDYT @Mrtenz?

Comment (Mrtenz) [reply packages/snaps-sdk/src/jsx/validation.ts:249]: I think it should, as we'd want to style it differently for `primary` and `destructive` buttons potentially.

Comment (FrederikBolding) [reply packages/snaps-sdk/src/jsx/validation.ts:249]: Even if we wouldn't right now, it makes it easy to do it in the future

Comment (eriknson) [reply packages/snaps-sdk/src/jsx/validation.ts:249]: Ye, I think all states (primary, disabled, destructive) should be able to load, initially with same spinner (inheriting the color of the state)

File: packages/snaps-sdk/src/jsx/components/form/Button.ts [hunk 2]
```diff
@@ -16,6 +16,7 @@ import type { ImageElement } from '../Image';
  * @property variant - The variant of the button, i.e., `'primary'` or
  * `'destructive'`. Defaults to `'primary'`.
  * @property disabled - Whether the button is disabled. Defaults to `false`.
+ * @property loading - Whether the button is in loading state. Defaults to `false`.
```
Comment (FrederikBolding) [inline packages/snaps-sdk/src/jsx/components/form/Button.ts:19]:
```suggestion
 * @property loading - Whether the button is loading. Defaults to `false`.
```

**Review by FrederikBolding — APPROVED** [review]

**Pull request merged**